core/server/server.js

```diff
@@ -173,22 +206,25 @@ class Server {
    * Badge Server Constructor
    *
    * @param {object} config Configuration object read from config yaml files
-   *    by https://www.npmjs.com/package/config and validated against
-   *    publicConfigSchema and privateConfigSchema
+   * by https://www.npmjs.com/package/config and validated against
+   * publicConfigSchema and privateConfigSchema
+   * @param {object} instanceMetadata Metadata of a running server instance
+   * @param {string} instanceMetadata.id Identifier of a running server instance
+   * @param {string} instanceMetadata.env Environment of a running server instance
+   * @param {string} instanceMetadata.hostname Hostname of a running server instance
    * @see https://github.com/badges/shields/blob/master/doc/production-hosting.md#configuration
    * @see https://github.com/badges/shields/blob/master/doc/server-secrets.md
    */
-  constructor(config) {
+  constructor(config, instanceMetadata) {
     const publicConfig = Joi.attempt(config.public, publicConfigSchema)
-    let privateConfig
-    try {
-      privateConfig = Joi.attempt(config.private, privateConfigSchema)
-    } catch (e) {
-      const badPaths = e.details.map(({ path }) => path)
-      throw Error(
-        `Private configuration is invalid. Check these paths: ${badPaths.join(
-          ','
-        )}`
+    const privateConfig = this.validatePrivateConfig(
+      config.private,
+      privateConfigSchema
+    )
+    if (publicConfig.metrics.influx && publicConfig.metrics.influx.enabled) {
+      this.validatePrivateConfig(
+        config.private,
+        privateMetricsInfluxConfigSchema
```

Comment:
Why does this need to be validated again?

Reply:
I want to require `username` and `password` for influx metrics only if influx metrics are enabled. The private config schema and the public config schema are two separate schemas and I did't find a different way to do it.

Reply:
But we do not have to require the `username` and the `password` in the influx config. Maybe it's not worth to keep such complicated code for this validation.

Reply:
Ahh gotcha. Why don't you make the call, and if you think it's worth keeping, add a comment in the code explaining why it's implemented the way it is.

Reply:
I added a comment with an explanation.